lib/phonebook/providers/association_kinds_provider.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:myecl/auth/providers/openid_provider.dart';
+import 'package:myecl/phonebook/class/association_kinds.dart';
+import 'package:myecl/phonebook/repositories/association_repository.dart';
+import 'package:myecl/tools/providers/single_notifier.dart';
+import 'package:myecl/tools/token_expire_wrapper.dart';
+
+class AssociationKindsNotifier extends SingleNotifier<AssociationKinds> {
+  final AssociationRepository associationRepository = AssociationRepository();
+  AssociationKindsNotifier({required String token})
+      : super(const AsyncValue.loading()) {
+    associationRepository.setToken(token);
+  }
+
+  void setKind(AssociationKinds i) {
+    state = AsyncValue.data(i);
+  }
+
+  Future<AsyncValue<AssociationKinds>> loadAssociationKinds() async {
+    AsyncValue<AssociationKinds> result =
+        await load(() async => associationRepository.getAssociationKinds());
```

Comment:
You can just write load(associationRepository.getAssociationKinds), which will work the same and is cleaner

Reply:
Fix in next commit